### PR TITLE
[UpdateUtils]

### PR DESCRIPTION
**Pull request by Just-Jojo**
## Type
- [ ] Bugfix
- [ ] Enhancement
- [ ] New Feature
- [x] New cog

## Description of the changes
Adds update utils, a cog designed for updating my utils :kappa:

### CODE REVIEW

File: updateutils/update.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        self.bot = bot
+        self.session = aiohttp.ClientSession()
+        sep = os.path.sep
+        # This section is a modified version of Jack's shell's `get_env` function
+        # Which is copyrighted under the Apache-2.0 License.
+        # https://github.com/jack1142/JackCogs/blob/v3/shell/utils.py#L30
+        if sys.platform == "win32":
+            self.path = f"{sys.prefix}{sep}Scripts{sep}python.exe"
+        else:
+            self.path = f"{sys.prefix}{sep}bin{sep}python3.8"
```
Comment (Kreusada) [inline updateutils/update.py:41]:
```suggestion
        self.path = Path(sys.executable).resolve()
```

File: updateutils/update.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
+# Copyright (c) 2021 - Jojo#7791
+# Licensed under MIT
+
+import asyncio
+import asyncio.subprocess as asp
+import os
+import sys
+from datetime import datetime
+
+import aiohttp
```
Comment (Kreusada) [inline updateutils/update.py:10]:
```suggestion
from datetime import datetime
from pathlib import Path

import aiohttp
```

File: updateutils/update.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        """Attempt to update your version of Jojo's utils"""
+
+        latest_version = await self._get_version()
+        if jojoutils_version == latest_version:
+            await ctx.send("Your version of Jojo's utils is up to date")
+            return
+        pred = MessagePredicate.yes_or_no(ctx)
+        msg = await ctx.send(
+            "This is the command that will be ran. Are you okay with this (input y/n)?"
+            f"\n\n`<path to your venv's python> {self.command}`"
```
Comment (Kreusada) [inline updateutils/update.py:151]:
```suggestion
            f"\n\n`{self.path} {self.command}`"
```
I feel like this should contain the path

File: updateutils/update.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+            msg = await self.bot.wait_for("message", check=pred, timeout=15.0)
+        except asyncio.TimeoutError:
+            pass
+        if not pred.result:
+            await ctx.send("That's fine. If you ever want to update you can run that command in your console.")
+            return
+        try:
+            await msg.add_reaction("\N{WHITE HEAVY CHECK MARK}")
+        except Exception:
+            pass
```
Comment (Kreusada) [inline updateutils/update.py:163]:
```suggestion
        await ctx.tick()
```

File: updateutils/info.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    "author": [
+        "Jojo#7791"
+    ],
+    "required_cogs": {},
+    "requirements": [
+        "git+https://github.com/Just-Jojo/jojoutils.git"
+    ],
+    "tags": ["utils", "dev"],
+    "min_bot_version": "3.4.6",
+    "hidden": false,
```
Comment (Kreusada) [inline updateutils/info.json:16]:
```suggestion
    "hidden": true,
```
If you're going for approved, this kind of cog *should* remain hidden from the index

File: updateutils/update.py (hunk 1 above)
Comment (Kreusada) [reply updateutils/update.py:41]: ``sys.executable`` is essentially exactly the same as what you're doing here, fyi

**Review by Kreusada — COMMENTED** [review]
A few minor changes but otherwise looking good!

File: updateutils/update.py (hunk 4 above)
Comment (Just-Jojo) [reply updateutils/update.py:163]: I'm gonna keep this as is because I'd rather have the `y` message ticked instead of the command ticked.

**Comment by Just-Jojo** [comment]
Thanks! The requested changes have been made (bar one lol)

**Pull request merged**